### PR TITLE
Catch more exceptions during install & display the error intead of a JS error

### PR DESCRIPTION
**Pull request by rGaillard**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.4.x
| Description?  | Some exceptions are not catched during the installation which lead to a 500 error instead of a message displayed. Also, in some case, the error message is not displayed (JS error).
| Type?         | bug fix 
| Category?     | IN
| BC breaks?    | no
| Deprecations? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9271)

### CODE REVIEW

**Comment by PierreRambaud** [comment]
Ping @Quetzacoalt91 @tomlev

**Review by tomlev — DISMISSED** [review]

File: install-dev/theme/js/process.js [hunk 1]
```diff
@@ -177,7 +177,7 @@ function install_error(step, errors)
 			list_errors = [];
 			list_errors[0] = errors;
 		} else if ($.type(list_errors) == 'array') {
-			list_errors = list_errors[0];
+			list_errors = [list_errors[0]];
```
Comment (Quetzacoalt91) [inline install-dev/theme/js/process.js:180]: Aren't we missing additional error messages by keeping only the first row?

Comment (rGaillard) [reply install-dev/theme/js/process.js:180]: It was like that since a while... but currently throw a JS error on the each below.

**Review by Quetzacoalt91 — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
Is there something for QA? Should I add the label?

**Review by eternoendless — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9271)* status: 1 of 3 files reviewed, 2 unresolved discussions (waiting on @Quetzacoalt91 and @rGaillard)

___
*[install-dev/theme/js/process.js, line 180 at r1](https://reviewable.io/reviews/prestashop/prestashop/9271#-LGilTDX-IGqGHpyfwGn-r1-180:-LHCnFU509ILuzSy9FWZ:b9tc50y) ([raw file](https://github.com/prestashop/prestashop/blob/f4e21bb4a3474de4eb5c8a3c2893a7ac0b3c6b59/install-dev/theme/js/process.js#L180)):*
<details><summary><i>Previously, rGaillard (Rémi Gaillard) wrote…</i></summary><blockquote>

It was like that since a while... but currently throw a JS error on the each below. 
</blockquote></details>

You shouldn't truncate the list of errors. I thought you said the problem was trying to iterate over a string?

I think you should drop the `else if` block altogether instead.
___
*[src/PrestaShopBundle/Install/Install.php, line 516 at r2](https://reviewable.io/reviews/prestashop/prestashop/9271#-LHCncBHDj81I4lpl8Ls:-LHCncBI7kf44AjPY7eq:bbdflt0) ([raw file](https://github.com/prestashop/prestashop/blob/28cf6b64652c68a1ef96a5d9c0f37cdf038db7cb/src/PrestaShopBundle/Install/Install.php#L516)):*
> ```PHP
>             }
> 
>         } catch (PrestashopInstallerException $e) {
> ```

Why not catch `Exception`? That way you'll be sure to "catch them all"

**Review by eternoendless — COMMENTED** [review]
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9271)* status: all files reviewed, 2 unresolved discussions (waiting on @Quetzacoalt91 and @rGaillard)

**Review by rGaillard — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9271)* status: all files reviewed, 2 unresolved discussions (waiting on @Quetzacoalt91 and @rGaillard)

___
*[install-dev/theme/js/process.js, line 180 at r1](https://reviewable.io/reviews/prestashop/prestashop/9271#-LGilTDX-IGqGHpyfwGn-r1-180:-LHCpBmr2uG_xtzGzT1z:b-6uptzr) ([raw file](https://github.com/prestashop/prestashop/blob/f4e21bb4a3474de4eb5c8a3c2893a7ac0b3c6b59/install-dev/theme/js/process.js#L180)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

You shouldn't truncate the list of errors. I thought you said the problem was trying to iterate over a string?

I think you should drop the `else if` block altogether instead.
</blockquote></details>

Because i don't want to break the current behavior.  list_errors[0] is always a string.
___
*[src/PrestaShopBundle/Install/Install.php, line 516 at r2](https://reviewable.io/reviews/prestashop/prestashop/9271#-LHCncBHDj81I4lpl8Ls:-LHCogq0-58QselWy8hP:b-retnt5) ([raw file](https://github.com/prestashop/prestashop/blob/28cf6b64652c68a1ef96a5d9c0f37cdf038db7cb/src/PrestaShopBundle/Install/Install.php#L516)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Why not catch `Exception`? That way you'll be sure to "catch them all"
</blockquote></details>

This is the case in the process.php file where all exceptions are catched.

**Review by eternoendless — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9271)* status: all files reviewed, 1 unresolved discussion (waiting on @Quetzacoalt91)

___
*[install-dev/theme/js/process.js, line 180 at r1](https://reviewable.io/reviews/prestashop/prestashop/9271#-LGilTDX-IGqGHpyfwGn-r1-180:-LHCulet4x2Tm8Wr-iWE:bjeftl) ([raw file](https://github.com/prestashop/prestashop/blob/f4e21bb4a3474de4eb5c8a3c2893a7ac0b3c6b59/install-dev/theme/js/process.js#L180)):*
<details><summary><i>Previously, rGaillard (Rémi Gaillard) wrote…</i></summary><blockquote>

Because i don't want to break the current behavior.  list_errors[0] is always a string.
</blockquote></details>

Ok we'll see about that later

**Comment by eternoendless** [comment]
Thank you @rGaillard

**Pull request merged**